packages/playground/src/App.vue

```diff
@@ -253,6 +271,42 @@ onMounted(async () => {
   openProfile.value = true;
 });
 
+watch(hasClient, () => setTimeouts());
+
+async function setTimeouts() {
+  if (!localStorage.getItem(LocalStorageSettingsKey.TIMEOUT_QUERY_KEY)) {
+    localStorage.setItem(LocalStorageSettingsKey.TIMEOUT_QUERY_KEY, `${window.env.TIMEOUT / 1000}`);
+  } else {
+    window.env.TIMEOUT = +localStorage.getItem(LocalStorageSettingsKey.TIMEOUT_QUERY_KEY)! * 1000;
+  }
+
+  const client = gridStore.client as GridClient;
+
+  const localStorageDeploymentTimeout = localStorage.getItem(LocalStorageSettingsKey.TIMEOUT_DEPLOYMENT_KEY);
+  if (!localStorageDeploymentTimeout) {
+    if (client) {
+      const clientOptions = client.clientOptions;
+      if (clientOptions) {
+        const deploymentTimeoutMinutes = clientOptions.deploymentTimeoutMinutes;
+        if (deploymentTimeoutMinutes) {
+          localStorage.setItem(LocalStorageSettingsKey.TIMEOUT_DEPLOYMENT_KEY, `${+deploymentTimeoutMinutes * 60}`);
+        }
+      }
+    }
+  } else {
+    if (client) {
+      const clientOptions = client.clientOptions;
+      if (clientOptions) {
+        const deploymentTimeoutMinutes = clientOptions.deploymentTimeoutMinutes;
+        if (deploymentTimeoutMinutes) {
+          client.clientOptions.deploymentTimeoutMinutes = +localStorageDeploymentTimeout / 60;
+          await client.connect();
+        }
+      }
+    }
+  }
+}
```

Comment:
```suggestion
async function setTimeouts() {
  if (!localStorage.getItem(LocalStorageSettingsKey.TIMEOUT_QUERY_KEY)) {
    localStorage.setItem(LocalStorageSettingsKey.TIMEOUT_QUERY_KEY, `${window.env.TIMEOUT / 1000}`);
  } else {
    window.env.TIMEOUT = +localStorage.getItem(LocalStorageSettingsKey.TIMEOUT_QUERY_KEY)! * 1000;
  }

  const client = gridStore.client as GridClient;

  const localStorageDeploymentTimeout = localStorage.getItem(LocalStorageSettingsKey.TIMEOUT_DEPLOYMENT_KEY);
  if (client) {
    if (!localStorageDeploymentTimeout) {
      const deploymentTimeoutMinutes = client.clientOptions.deploymentTimeoutMinutes;
      if (deploymentTimeoutMinutes) {
        localStorage.setItem(LocalStorageSettingsKey.TIMEOUT_DEPLOYMENT_KEY, `${+deploymentTimeoutMinutes * 60}`);
      } else {
        client.clientOptions.deploymentTimeoutMinutes = +localStorageDeploymentTimeout! / 60;
        await client.connect();
      }
    }
  }
}
```

Reply:
this doesn't work as clientoptions is not seen if not checked for before setting the timeout in the client

Reply:
This should do.
```
async function setTimeouts() {
  if (!localStorage.getItem(LocalStorageSettingsKey.TIMEOUT_QUERY_KEY)) {
    localStorage.setItem(LocalStorageSettingsKey.TIMEOUT_QUERY_KEY, `${window.env.TIMEOUT / 1000}`);
  } else {
    window.env.TIMEOUT = +localStorage.getItem(LocalStorageSettingsKey.TIMEOUT_QUERY_KEY)! * 1000;
  }

  const client = gridStore.client as GridClient;
  const clientOptions = client.clientOptions;
  const localStorageDeploymentTimeout = localStorage.getItem(LocalStorageSettingsKey.TIMEOUT_DEPLOYMENT_KEY);
  const deploymentTimeoutMinutes = clientOptions.deploymentTimeoutMinutes;

  if (client && clientOptions && deploymentTimeoutMinutes) {
    if (!localStorageDeploymentTimeout) {
      localStorage.setItem(LocalStorageSettingsKey.TIMEOUT_DEPLOYMENT_KEY, `${+deploymentTimeoutMinutes * 60}`);
    } else {
      client.clientOptions.deploymentTimeoutMinutes = +localStorageDeploymentTimeout! / 60;
      await client.connect();
    }
  }
}
```